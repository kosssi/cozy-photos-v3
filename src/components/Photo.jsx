import styles from '../styles/photo'

import React from 'react'

const STACK_FILES_DOWNLOAD_PATH = 'http://cozy.local:8080/files/download'

const Photo = ({ photo }) => {
  return (
    <img
      className={styles['pho-photo']}
      src={`${STACK_FILES_DOWNLOAD_PATH}/${photo._id}`}
    />
  )
}

export default Photo
