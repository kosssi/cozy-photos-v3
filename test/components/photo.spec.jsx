'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'
import Photo from '../../src/components/Photo'

const photoObject = {
  _id: '33dda00f0eec15bc3b3c59a615001ac8',
  created_at: '0001-01-01T00:00:00Z',
  name: 'MonImage.jpg',
  size: '150000',
  updated_at: '0001-01-01T00:00:00Z'
}

describe('Photo component', () => {
  it('should render correctly an image according a photo object', () => {
    const component = shallow(
      <Photo photo={photoObject} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
