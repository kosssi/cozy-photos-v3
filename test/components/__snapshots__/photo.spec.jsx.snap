exports[`Photo component should render correctly an image according a photo object 1`] = `
<img
  className="pho-photo"
  src="http://cozy.local:8080/files/download/33dda00f0eec15bc3b3c59a615001ac8" />
`;
